### ISSUE TITLE
AttributeError: 'ShapedArray' object has no attribute 'val'

### ISSUE THREAD

**Issue by Kangfei** [google/neural-tangents#74]
I'm running the simple demo as facing the array problem:

`from jax import random
import jax.numpy as np
from jax.api import jit
from neural_tangents import stax

init_fn, apply_fn, kernel_fn = stax.serial(
    stax.Dense(512), stax.Relu(),
    stax.Dense(512), stax.Relu(),
    stax.Dense(1)
)

key1, key2 = random.split(random.PRNGKey(1))
x1 = random.normal(key1, (10, 100))
x2 = random.normal(key2, (20, 100))

kernel = kernel_fn(x1, x2, 'nngp')`

/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/jax/lax/lax.py:6190: UserWarning: Explicitly requested dtype <class 'jax.numpy.lax_numpy.float64'> requested in astype is not available, and will be truncated to dtype float32. To enable more dtypes, set the jax_enable_x64 configuration option or the JAX_ENABLE_X64 shell environment variable. See https://github.com/google/jax#current-gotchas for more.
  warnings.warn(msg.format(dtype, fun_name , truncated_dtype))
Traceback (most recent call last):
  File "/home/kfzhao/PycharmProjects/NNGP/test.py", line 25, in <module>
    kernel = kernel_fn(x1, x2, 'nngp')
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/utils/utils.py", line 84, in h
    return g(*args, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/utils/utils.py", line 118, in getter_fn
    fn_out = fn(*canonicalized_args, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 2935, in kernel_fn_any
    **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 2852, in kernel_fn_x1
    out_kernel = kernel_fn(kernel, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/utils/utils.py", line 84, in h
    return g(*args, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 185, in new_kernel_fn
    return kernel_fn(k, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 318, in kernel_fn
    k = f(k, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/utils/utils.py", line 84, in h
    return g(*args, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/utils/utils.py", line 118, in getter_fn
    fn_out = fn(*canonicalized_args, **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 2928, in kernel_fn_any
    **kwargs)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 2844, in kernel_fn_kernel
    return _set_shapes(init_fn, kernel, out_kernel)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 2780, in _set_shapes
    shape1 = _propagate_shape(init_fn, in_kernel.shape1)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 2770, in _propagate_shape
    out_shape = tree_map(lambda x: int(x.val), out_shape)
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/jax/tree_util.py", line 164, in tree_map
    return treedef.unflatten(map(f, leaves))
  File "/home/kfzhao/.conda/envs/ntk/lib/python3.7/site-packages/neural_tangents/stax.py", line 2770, in <lambda>
    out_shape = tree_map(lambda x: int(x.val), out_shape)
AttributeError: 'ShapedArray' object has no attribute 'val'

Same problem also occurs when I'm running  the examples/infinite_fcn.py.  Here's some of my package configs on py 3.7.9:
jax                      0.2.1
jaxlib                   0.1.55 (for cuda 10.1)
numpy                    1.18.5
scipy                    1.5.2
Does someone have the same problem?

**Comment by romanngg** [comment]
Looks like old neural tangents code, could you try installing from head, i.e.
```
git clone https://github.com/google/neural-tangents; cd neural-tangents
pip install -e .
```
instead of pypi? I don't have access to my machine now, will update the pypi package next week!

**Comment by Kangfei** [comment]
> Looks like old neural tangents code, could you try installing from head, i.e.
> 
> ```
> git clone https://github.com/google/neural-tangents; cd neural-tangents
> pip install -e .
> ```
> 
> instead of pypi? I don't have access to my machine now, will update the pypi package next week!

Just tried installing from source and the problem is fixed. Thanks!

Best, Kangfei

**Comment by romanngg** [comment]
Fixed in [v0.3.4](https://github.com/google/neural-tangents/releases/tag/v0.3.4)!
